### DISCUSSION TITLE
VSD

### DISCUSSION THREAD

**Discussion by sofia-bg** [UNC-Project-Open-AAC/OS-DPI#90] [Help]
@gbishop I played around a bit with trying to "link" several VSDs to no avail. At one point added a grid with four buttons to imitate "tabs" with a changevsd column and tried to use role and changevsd to change the VSD when a button in the grid was selected but could not get to work. Any ideas? 

[playground VSD OSDPI file here](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/fff26a30ba73ce980a73e7ad1625b652c62c3b3f/Designs%20In%20Progress/vsd/Play_VSD.xlsx)

Can see in data have [playground](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/fff26a30ba73ce980a73e7ad1625b652c62c3b3f/Designs%20In%20Progress/vsd/playground.jpeg), [bedroom](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/fff26a30ba73ce980a73e7ad1625b652c62c3b3f/Designs%20In%20Progress/vsd/bedroom.jpeg), [picnic](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/fff26a30ba73ce980a73e7ad1625b652c62c3b3f/Designs%20In%20Progress/vsd/picnic.jpeg), [classroom](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/fff26a30ba73ce980a73e7ad1625b652c62c3b3f/Designs%20In%20Progress/vsd/classroom.png) images.

**Comment by gbishop** [comment]
It should work just like a grid where the filters govern what is currently
displayed. I can't look right now.

**Comment by gbishop** [comment]
I don't see the OSDPI file. Only the spreadsheet and images.

**Reply by sofia-bg** [reply]
Linked wrong file - try [this](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/f5f3b626b9927743567bc8a287288052435eb127/Designs%20In%20Progress/vsd/puppies_vsd.osdpi)

**Comment by sofia-bg** [comment]
@gbishop though I am going to work on it more today - I think I understand what I need to do differently

**Reply by sofia-bg** [reply]
@gbishop can you check this one out when you have a chance? Trying to use a grid like "tabs" to switch between VSD but not getting it to work. Tried init with a new image to see if I could force it to change to one of the others but no luck. 

[VSD play v2](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/5fcaae58fa1a9a4a2d6abb2ff43817888b229e1a/Designs%20In%20Progress/vsd/playground_vsd_v2.osdpi)

**Reply by gbishop** [reply]
In your layout for the VSD filters you've got

#vsd equals $image

You needed a capital I on that image. It is $Image. 

We need better error checking than we have. It should have flagged that expression.

**Reply by sofia-bg** [reply]
Thank you! I kept messing with the #vsd equals and the actions and it never worked, but always triggered and must have always had some sort of mismatch between the two. 

Will work on putting a coordinates in for the others so we can share should we want to at community meeting.

**Reply by gbishop** [reply]
Also, you had several issues in the spreadsheet. Look at the format in the one I sent you via email. 

The image lines need to be separate from the box coordinate lines and every line needs the values set so that the filters will pick them up.

**Reply by sofia-bg** [reply]
Hmm... this one works for me just fine though? or is it not working on your end? 

I have a mix of image on every line and not and all my hotspots are working. But will check the spreadsheet you sent and can reformat my own. 

[v4](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/f336f0b972f6faabf4b191d3fb3da0dec3d134fd/Designs%20In%20Progress/vsd/playground_vsd_v4.osdpi)

**Reply by gbishop** [reply]
I guess you can do it that way but the implication to a reader might be that the image could be different on every line and that isn't true. 

My intention was that you had a row specifying the image and then other rows specifying the boxes.

**Reply by sofia-bg** [reply]
That makes sense, easy enough to change. Just wasn't sure and went conservative with just putting information everywhere.

**Comment by gbishop** [comment]
You need pictures for the others also. And look at the spreadsheet I made
several changes
